### DISCUSSION TITLE
Accessing 'setup' before parent setup is executed

### DISCUSSION THREAD

**Discussion by LaravelFreelancerNL** [pestphp/pest#429] [Q&A]
I'm building a Laravel ArangoDB database driver which requires transactions to start with a list of used collections (tables).

So I need to set this information before the RefreshDatabase or DatabaseTransaction traits are used.

With phpunit I can solve this by overriding the beginDatabaseTransaction method in the traits and set a property on TestCase in setup before calling the parent:
```php
    protected function setUp(): void
    {
        $this->setTransactionCollections([
            'write' => ['pages']
        ]);

        parent::setUp();
    }
```

From what I understand beforeEach is the equivolant of setup. So I've tried:
```php 
beforeEach()->setTransactionCollections([
    'write' => ['pages']
]);```
 However this doesn't work. The set function is called after the transaction is initiated.

 Is there a way to hook the required information into the procedure to have it available in time?

**Comment by nunomaduro** [comment]
You need to create a normal test class, and use `uses('YourTestClass::class')` in your test.

**Reply by LaravelFreelancerNL** [reply]
Thanks for the quick reply!

That will work, however it would create a test class for each pest file that uses transactions. At least if you want to register the specific set of tables used in the test file.

However you've given me an idea: just write one TestCase (or a limited set) and register all tables in there.
I've just tried it out and this seems to work. And as parallel testing uses a database for each simultaneously running test there shouldn't be a problem there either(once I add support for it). So it is good enough for now.

Both options are a bit ugly though. So it would be nice to be able to hook into several parts of the createapplication/setup process at some point.
